src/server/active_tran_server.cpp

```diff
@@ -36,6 +36,25 @@ active_tran_server::uses_remote_storage () const
   return m_uses_remote_storage;
 }
 
+MVCCID
+active_tran_server::get_oldeset_active_mvccid_from_page_server () const
+{
+  std::string response_message;
+  const int error_code = send_receive (tran_to_page_request::GET_OLDEST_ACTIVE_MVCCID, std::string (), response_message);
+  if (error_code != NO_ERROR)
+    {
+      ASSERT_ERROR ();
+      return error_code;
+    }
+
+  MVCCID oldest_mvccid = MVCCID_NULL;
+  std::memcpy (&oldest_mvccid, response_message.c_str (), sizeof (oldest_mvccid));
+
+  assert (MVCCID_IS_VALID (oldest_mvccid));
+
+  return oldest_mvccid;
```

Comment:
I think that the exception handling codes are needed. The oldest_mvccid never be less than MVCCID_FIRST.

Reply:
It can be `MVCCID_ALL_VISIBLE`, actually. I did the check with `assert (MVCCID_IS_VALID (oldest_mvccid));`, but you mean we should handle the case?

Reply:
Oh. Sorry I mean the MVCCID_ALL_VISIBLE iso the MVCCID_FIRST. There is a room for 1 and 2. I worry about someone change the PS mechanism to can send 1 or 2 later. At that time this function can not filter even if wrong values.

Reply:
> There is a room for 1 and 2.

I understand the concern, then how about `assert (MVCCID_IS_NORMAL (oldest_mvccid) || MVCCID_ALL_VISIBLE (oldest_mvccid))`. I think we can add some comments here for each value, which is described where it is sent, though.

>At that time this function can not filter even if wrong values.

Do you mean we should more than assert?

Reply:
OK. I aggree your suggestion about assert.

> Do you mean we should more than assert?

I think that this function should have an exception handling code(kind of error code, error message) when it receive un-expected value from PS. But it is enough now to add new assert condition. consider this error handling mechanish later.

Reply:
OK, we can do as the above error handling:
```
  if (error_code != NO_ERROR)
     {
       return MVCCID_NULL;
     }
```
But, for that, I think we should define the error code. Let's go with just assertion for the time being.